src/Compilers/CSharp/Portable/FlowAnalysis/DataFlowPass.cs

```diff
@@ -334,20 +334,25 @@ protected void Analyze(ref bool badRegion, DiagnosticBag diagnostics)
             }
         }
 
-        private void CheckCaptured(Symbol variable)
+        private void CheckCaptured(Symbol variable, Symbol underlyingVariable = null)
```

Comment:
Consider adding a comment what is the underlyingVariable and when it should be passed in.